portable/GCC/ARM_CRx_MPU/portmacro_asm.h

```diff
@@ -0,0 +1,515 @@
+ *
+ *
+ * For more information about the FPU, FPSCR, and FPRs please reference ARM's website:
+ * https://developer.arm.com/documentation/den0042/a/Floating-Point
+ *
+ * Additional information related to the Cortex R4-F's FPU Implementation:
+ * https://developer.arm.com/documentation/ddi0363/e/fpu-programmer-s-model
+ *
+ * Additional information related to the Cortex R5-F's FPU Implementation:
+ * https://developer.arm.com/documentation/ddi0460/d/FPU-Programmers-Model
+ *
+ * Additional information related to the ArmV7-R CPSR
+ * https://developer.arm.com/documentation/ddi0406/cb/Application-Level-Architecture/Application-Level-Programmers--Model/The-Application-Program-Status-Register--APSR-?lang=en
+ *
+ * Additional information related to the GPRs:
+ * https://developer.arm.com/documentation/ddi0406/cb/System-Level-Architecture/The-System-Level-Programmers--Model/ARM-processor-modes-and-ARM-core-registers/ARM-core-registers?lang=en
+ *
+ */
+
+/** @brief The length in ulContext for the General Purpose Registers in bytes
+ * @note There are 13 GPRs, R0-R12 each is 32 bits, so 13 registers * 4 Bytes each
+ */
+#define portGPR_LENGTH                       ( 15U * 4U )
+
+/** @brief The length in ulContext for all the registers in a context
+ * @note There are the 13 GPRs, the Stack Pointer, and the Link Register
+ */
+#define portREGISTER_CONTEXT_LENGTH          ( ( 16 * 4U ) )
+
+/** If you KNOW that your system will not utilize the FPU in any capacity
+ * you can set portENABLE_FPU to 0, which will reduce the per-task RAM usage
+ * by ( 32 FPRs + 32 bit FPSCR ) * 4 bytes per register = 132, or 0x84, Bytes Per Task
+ * BE CAREFUL DISABLING THIS: Certain APIs will try and optimize themselves
+ * by using the FPRs. If the FPU context is not saved and this happens it could be
+ * exceedingly difficult to debug why a strcpy() or other similar function
+ * seems to randomly fail.
+ */
+#ifndef configENABLE_FPU
+    #define configENABLE_FPU 0
+#endif
+
+/** @brief Mark if the Floating Point Registers will be saved.
+ * @ingroup Task Context
+ * @note When using the FPU, we must save additional registers into the task's context
+ * These consist of the Floating Point Status and Control Register (FPSCR),
+ * As well as the Floating Point Registers (FPRs)
+ * Task Context which is stored in ulContext in order, consists of:
+ * ulContext[ 0 ]:      Critical Nesting Count: ulCriticalNesting
+ * ulContext[ 1 ]:      Floating Point Status and Control Register
+ * ulContext[ 2 - 33 ]: Floating Point Registers: S0-S31
+ * ulContext[ 34 - 46 ]:  General Purpose Registers: R0-R12
+ * ulContext[ 47 ]:     Stack Pointer
+ * ulContext[ 48 ]:     Link Register
+ * ulContext[ 49 ]:     Program Counter
+ * ulContext[ 50 ]:     Current Program Status and Control Register
+ */
+#define portENABLE_FPU configENABLE_FPU
+
+#if( portENABLE_FPU == 1 )
+    /** @brief Length of a Task's Register Context when using an FPU. */
+    #define MAX_CONTEXT_SIZE 50U
```

Comment:
Looks like there is a mismatch between the table above and the MAX_CONTEXT_SIZE

Reply:
Value increased